app/controllers/api/v2/puppet_hosts_controller.rb

```diff
@@ -1,27 +0,0 @@
-module Api
-  module V2
-    class PuppetHostsController < V2::BaseController
-      include Api::Version2
-      include HostsControllerExtension
-
-      check_permissions_for ['puppetrun']
-
-      api :PUT, "/hosts/:id/puppetrun", N_("Force a Puppet agent run on the host")
```

Comment:
Would it make sense to keep this but return a 501 Not Implemented as compatibility?

Reply:
I'd agree, or even deprecating it and run REX task if REX is present.